### PR TITLE
BUGFIX: Trim leading and tailing whitespaces

### PR DESCRIPTION
**Pull request by markusguenther**
Whitespaces like a tab can lead to a 404 error if a URL has been pasted to a `LinkEditor` or `LinkInput`.

As data sources, assets, URLs and slugs should not start with a whitespace, we can remove them.

Fixes: #3859

### CODE REVIEW

**Comment by markusguenther** [comment]
If I missed something, please let me know. Also thought about an editor option to trim that, but that would be more a feature than a bug fix then.


https://github.com/user-attachments/assets/9c247eeb-85ab-4918-b26f-1870df594bd4

File: packages/neos-ui-editors/src/Library/LinkInput.js [hunk 1]
```diff
@@ -149,6 +149,10 @@ export default class LinkInput extends PureComponent {
     }
 
     handleSearchTermChange = searchTerm => {
+        // trim leading and trailing whitespace as it can cause issues
+        // with the further processing
+        searchTerm.trim()
```
Comment (Sebobo) [inline packages/neos-ui-editors/src/Library/LinkInput.js]: Could there be a case in which `searchTerm` is null or undefined?

Comment (markusguenther) [reply packages/neos-ui-editors/src/Library/LinkInput.js]: Right, that should be checked!

**Comment by markusguenther** [comment]
The acceptance tests will run when https://github.com/neos/neos-ui/pull/3855 has been merged. At the moment they are broken.

File: packages/neos-ui-editors/src/Library/LinkInput.js (hunk 1 above)
Comment (markusguenther) [reply packages/neos-ui-editors/src/Library/LinkInput.js]: Normally, it should be always a string, but it is better to prevent a potential error

**Review by Sebobo — APPROVED** [review]
Looks good to me by reading, thx!

**Comment by pKallert** [comment]
Thanks for having a look! 
I tried it out, but it only works when entering something by hand. Copy/pasting a URL with tab/space still saves with the space.

**Comment by markusguenther** [comment]
> Thanks for having a look! I tried it out, but it only works when entering something by hand. Copy/pasting a URL with tab/space still saves with the space.

Will test it again, but also used the Clipboard.

**Comment by markusguenther** [comment]
@pKallert Don't know why it worked before for me 🙈. Thanks for the test and review.

**Comment by markusguenther** [comment]
Ok , as the `handleSearchTermChange` is called on each keystroke we trim also spaces while entering text, so I now just trim the beginning at this point and when the Link should be applied manually we know that nothing should change anymore, so I can also trim the tailing whitespaces.

Hope that we now have a solution that works for all cases 🙈

**Comment by markusguenther** [comment]
Finally, tests are also passing 🥳

**Review by pKallert — APPROVED** [review]
Works like a charm, thank you!

**Pull request merged**